src/slurmdbd_ops.py

```diff
@@ -0,0 +1,395 @@
+
+    def install(self) -> bool:
+        """Install package using lib apt."""
+        package_installed = False
+
+        # Install the key.
+        if self._keyring_path.exists():
+            self._keyring_path.unlink()
+        self._keyring_path.write_text(UBUNTU_HPC_PPA_KEY)
+
+        # Add the repo.
+        repositories = apt.RepositoryMapping()
+        repositories.add(self._repo())
+
+        try:
+            apt.update()
+            apt.add_package([self._package_name])
+            package_installed = True
+        except apt.PackageNotFoundError:
+            logger.error(f"'{self._package_name}' not found in package cache or on system.")
+        except apt.PackageError as e:
+            logger.error(f"Could not install '{self._package_name}'. Reason: {e.message}")
+
+        return package_installed
+
+    def uninstall(self) -> None:
+        """Uninstall the package using libapt."""
+        if apt.remove_package(self._package_name):
+            logger.info(f"'{self._package_name}' removed from system.")
+        else:
+            logger.error(f"'{self._package_name}' not found on system.")
+
+        # Disable the ubuntu-hpc repo.
+        repositories = apt.RepositoryMapping()
+        repositories.disable(self._repo())
+
+        # Remove the key.
+        if self._keyring_path.exists():
+            self._keyring_path.unlink()
+
+    def upgrade_to_latest(self) -> None:
+        """Upgrade package to latest."""
+        try:
+            slurm_package = apt.DebianPackage.from_system(self._package_name)
+            slurm_package.ensure(apt.PackageState.Latest)
+            logger.info(f"Updated '{self._package_name}' to: {slurm_package.version.number}.")
+        except apt.PackageNotFoundError:
+            logger.error(f"'{self._package_name}' not found in package cache or on system.")
+        except apt.PackageError as e:
+            logger.error(f"Could not install '{self._package_name}'. Reason: {e.message}")
+
+    def version(self) -> str:
+        """Return the package version."""
+        try:
+            slurmdbd_vers = apt.DebianPackage.from_installed_package(self._package_name)
+        except apt.PackageNotFoundError:
+            logger.error(f"'{self._package_name}' not found on system.")
+        return slurmdbd_vers.version.number
+
+
+class SlurmdbdOpsManager:
+    """SlurmdbdOpsManager."""
+
+    def __init__(self):
+        """Set the initial attribute values."""
+        self._slurm_component = "slurmdbd"
+
+        self._slurm_state_dir = Path("/var/spool/slurmdbd")
+        self._jwt_rsa_key_file = self._slurm_state_dir / "jwt_hs256.key"
+        self._munge_key_path = Path("/etc/munge/munge.key")
+        self._slurmdbd_conf_dir = Path("/etc/slurm")
+        self._slurmdbd_conf_file = self._slurmdbd_conf_dir / "slurmdbd.conf"
+        self._slurmdbd_log_dir = Path("/var/log/slurm")
+        self._slurmdbd_log_file = self._slurmdbd_log_dir / "slurmdbd.log"
```

Comment:
I've noticed that when searching for usages of the attributes, I'm seeing that many of them are only ever read once or twice. I also see that some attributes - for example `_jwt_rsa_key_file` - are then assigned to a local variable within the method that they are called.

I think it would be simpler to create the `Path(...)` objects locally, on-the-fly within the methods that they are used. Since we're opinionated about where these files are going to be located, I don't think we need to make these values object attributes that can be redefined accidentally. Having a bunch of protected attributes that are only accessed once or twice adds additional code complexity where I think we could get away with hardcoded values.